Lib/test/test_asyncio/test_proactor_events.py

```diff
@@ -775,5 +779,168 @@ def test_stop_serving(self):
         self.assertFalse(future2.cancel.called)
 
 
+@unittest.skipIf(sys.platform != 'win32',
+                 'Proactor is supported on Windows only')
+class ProactorEventLoopUnixSockSendfileTests(test_utils.TestCase):
```

Comment:
Are these entirely new tests?  I thought we already had many functional tests for *nix.  Can we just reuse those?

Reply:
It was my try but I figured out that tests from `test_unix_events_py` not completely functional only, they use selector specific mocks for testing edge cases.
Proactor based implementation is different.

Reply:
Is it possible to have a common base class with functional tests and subclasses for selector- and proactor-loops with custom `setUp()` methods for mocking?

Having one source for sendfile tests would be super great and useful.

Reply:
I'll try to master something

Reply:
ping

Reply:
I'm on conference now, will have a time to work on tests at the beginning of the next week.

Reply:
Done